src/test/ui/suggestions/field-access.rs

```diff
@@ -0,0 +1,26 @@
+// run-rustfix
+#![allow(dead_code)]
+
+struct A {
+    b: B,
+}
+
+enum B {
+    Fst,
+    Snd,
+}
+
+fn main() {
+    let a = A { b: B::Fst };
+    if let B::Fst = a {};
+    //~^ ERROR mismatched types [E0308]
+    // note: you might have meant to use field `b` of type `B`
+    match a {
+        B::Fst => (),
+        B::Snd => (),
+    }
+    //~^^^ ERROR mismatched types [E0308]
+    // note: you might have meant to use field `b` of type `B`
+    //~^^^^ ERROR mismatched types [E0308]
+    // note: you might have meant to use field `b` of type `B`
```

Comment:
I feel that short messages are easier to follow when inlined. When running the tests these comments get striped out so they don't pollute the `.stderr` output.

```suggestion
    if let B::Fst = a {}; //~ ERROR mismatched types [E0308]
    //~^ HELP you might have meant to use field `b` of type `B`
    match a {
        B::Fst => (),  //~ ERROR mismatched types [E0308]
        //~^ HELP you might have meant to use field `b` of type `B`
        B::Snd => (),  //~ ERROR mismatched types [E0308]
        //~^ HELP you might have meant to use field `b` of type `B`
    }
```